v8/payloads/interactions.ts

```diff
@@ -0,0 +1,86 @@
+import type { RESTPostAPIWebhookWithTokenJSONBody } from '../rest';
+import type { MessageFlags } from './channel';
+import type { APIGuildMember } from './guild';
+
+export interface APIApplicationCommand {
+	id: string;
+	application_id: string;
+	name: string;
+	description: string;
+	options?: APIApplicationCommandOption[];
+}
+
+export interface APIApplicationCommandOption {
+	type: ApplicationCommandOptionType;
+	name: string;
+	description: string;
+	default?: boolean;
+	required?: boolean;
+	choices?: APIApplicationCommandOptionChoice[];
+	options?: APIApplicationCommandOption[];
+}
+
+export enum ApplicationCommandOptionType {
+	SUB_COMMAND = 1,
+	SUB_COMMAND_GROUP,
+	STRING,
+	INTEGER,
+	BOOLEAN,
+	USER,
+	CHANNEL,
+	ROLE,
+}
+
+export interface APIApplicationCommandOptionChoice {
+	name: string;
+	value: string | number;
+}
+
+export interface APIInteraction {
+	id: string;
+	type: InteractionType;
+	data?: APIApplicationCommandInteractionData;
+	guild_id: string;
+	channel_id: string;
+	member: APIGuildMember;
```

Comment:
The `user` on this member is always present.

```suggestion
	member: APIGuildMember & { user: APIUser; };
```

Can't add the import on top sadly.